lxd/storage/backend_lxd.go

```diff
@@ -1686,8 +1686,8 @@ func (b *lxdBackend) GetInstanceUsage(inst instance.Instance) (int64, error) {
 
 // SetInstanceQuota sets the quota on the instance's root volume.
 // Returns ErrInUse if the instance is running and the storage driver doesn't support online resizing.
-func (b *lxdBackend) SetInstanceQuota(inst instance.Instance, size string, op *operations.Operation) error {
-	logger := logging.AddContext(b.logger, log.Ctx{"project": inst.Project(), "instance": inst.Name(), "size": size})
+func (b *lxdBackend) SetInstanceQuota(inst instance.Instance, size string, fsSize string, op *operations.Operation) error {
```

Comment:
@stgraber how about calling this `vmStateSize` to indicate it is for VMs only? 

Also can you add a check that if this is non-empty for a normal filesystem volume it returns an error (before any resizing of either volumes is actioned), to avoid any unexpected usage in the future.

Finally, I think there are some helper functions you could use here, `vol.IsVMBlock()` which then only considers block volumes for instances and images, and not custom block volumes, and `vol.NewVMBlockFilesystemVolume()`.

So something like:

```go
        vol := b.newVolume(volType, contentVolume, volStorageName, nil)
        if !vol.IsVMBlock() && fsSize != "" {
           // error here
        }

        err = b.driver.SetVolumeQuota(vol, size, op)
	if err != nil {
		return err
	}

        if !vol.IsVMBlock() {
	        if fsSize == "" {
			fsSize = drivers.DefaultVMBlockFilesystemSize
		}

                fsVol := vol.NewVMBlockFilesystemVolume()
		err := b.driver.SetVolumeQuota(vol, fsSize, op)
		if err != nil {
			return err
		}
        }
```

Reply:
See below, but I think `NewVMBlockFilesystemVolume` should just accept `vmStateSize` and the logic to detect empty valued and fallback to `DefaultVMBlockFilesystemSize` should be in `NewVMBlockFilesystemVolume`.

Reply:
So we shouldn't fail if someone attempts to pass a vmStateSize with a non-VM.
This will happen when a project contains the config key.